src/components/OSCALCatalogGroup.js

```diff
@@ -18,57 +18,54 @@ export const OSCALControlList = styled(List)`
   padding-right: 2em;
 `;
 
-export default function OSCALCatalogGroup(props) {
+function OSCALCatalogGroupControl(props) {
   const [open, setOpen] = React.useState(false);
 
   const handleClick = () => {
     setOpen(!open);
   };
 
-  // Groups may not necessarily have an ID (it is not required per the spec);
-  // therefore, we need to be able to come up with a semi-constant ID. All
-  // groups will have a title. We can (poorly) hash that hopefully that will
-  // be good enough.
-  const groupKey = (group) => {
-    if (group.id) {
-      return group.id;
-    }
-    let hash = 7;
-    // eslint-disable-next-line no-restricted-syntax
-    for (const char of group.title) {
-      hash = 31 * hash + char.charCodeAt(0);
-    }
-    return hash;
-  };
+  const { control } = props;
 
   return (
     <div>
-      <ListItem key={groupKey(props.group)} button onClick={handleClick}>
-        <ListItemAvatar key={`${groupKey(props.group)}-avatar`}>
+      <ListItem button onClick={handleClick}>
+        <ListItemAvatar key={`${control.id}-avatar`}>
           <Avatar variant="rounded">
             <FolderIcon />
           </Avatar>
         </ListItemAvatar>
-        <ListItemText primary={props.group.title} />
+        <ListItemText
+          primary={`${control.id.toUpperCase()} ${control.title}`}
+        />
         {open ? <ExpandLess /> : <ExpandMore />}
       </ListItem>
+
       <Collapse in={open} timeout="auto" unmountOnExit>
-        <Box display="flex" justifyContent="flex-end">
-          <OSCALControlParamLegend />
-        </Box>
-        <OSCALControlList>
-          {props.group.groups?.map((innerGroup) => (
-            <OSCALCatalogGroup group={innerGroup} key={innerGroup.id} />
-          ))}
-          {props.group.controls?.map((control) => (
-            <OSCALControl
-              control={control}
-              childLevel={0}
-              key={`control-${control.id}`}
-            />
-          ))}
-        </OSCALControlList>
+        <OSCALControl
+          control={control}
+          childLevel={0}
+          key={`control-${control.id}`}
+        />
       </Collapse>
     </div>
   );
 }
+
+export default function OSCALCatalogGroup(props) {
+  return (
+    <div>
+      <Box display="flex" justifyContent="flex-end">
+        <OSCALControlParamLegend />
+      </Box>
+      <OSCALControlList>
+        {props.group.groups?.map((innerGroup) => (
+          <OSCALCatalogGroup group={innerGroup} key={innerGroup.id} />
```

Comment:
`id` may not be present on `group`s. That's what the `groupKey` function was for. We could just use `group.title`. But `groupKey` gave us a shorter number instead of a super long string.